realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -264,6 +290,86 @@ public boolean hasField(String fieldName) {
     }
 
     /**
+     * Set the value for the given field. This method will automatically try to convert numbers and
+     * booleans that are given as {@code String} to their appropriate type. E.g. {@code "10"} will be
+     * converted to {@code 10} if the field type is {@code int}.
+     *
+     * Using the typed setters will be faster than using this method.
+     *
+     * @throws IllegalArgumentException if field name doesn't exists or if the input value cannot be converted
+     * to the appropriate input type.
+     * @throws NumberFormatException if a String based number cannot be converted properly.
+     */
+    @SuppressWarnings("unchecked")
+    public void set(String fieldName, Object value) {
+        boolean isString = (value instanceof String);
+        String strValue = isString ? (String) value : null;
```

Comment:
should we check if the user passed a Date as String? maybe using `JsonUtils.stringToDate`